internal/report/generator.go

```diff
@@ -0,0 +1,102 @@
+package report
+
+import (
+	"bytes"
+	"text/template"
+	"time"
+)
+
+type ReportData struct {
+	ChannelName       string
+	WeekRange         string
+	TotalIncidents    int
+	CriticalIncidents int
+	MajorIncidents    int
+	MinorIncidents    int
+	Incidents         []IncidentSummary
+	AvgResponseTime   string
+	ResolutionRate    float64
+	TeamScore         float64
+	ActionItems       []string
+	GeneratedAt       string
+}
+
+type IncidentSummary struct {
+	Title    string
+	Severity string
+	Duration string
+	Status   string
+	Impact   string
+}
+
+type Generator struct {
+	template *template.Template
+}
+
+func NewGenerator() (*Generator, error) {
+	tmpl, err := template.ParseFiles("internal/templates/weekly_report.md")
```

Comment:
This only works because you run binary from that folder? To make binary portable we need to package the template with the binary. See how I do it in http.go